cloudformation-templates/pattern-1/pattern-1-with-puppet-cloudformation.template.yml

```diff
@@ -0,0 +1,603 @@
+  subnetacl3:
+      Type: 'AWS::EC2::SubnetNetworkAclAssociation'
+      Properties:
+        NetworkAclId: !Ref WSO2EnvNetworkAcl
+        SubnetId: !Ref WSO2EnvPublicSubnet2
+  gw1:
+    Type: 'AWS::EC2::VPCGatewayAttachment'
+    Properties:
+      VpcId: !Ref WSO2EnvVPC
+      InternetGatewayId: !Ref WSO2EnvInternetGateway
+  subnetroute1:
+    Type: 'AWS::EC2::SubnetRouteTableAssociation'
+    Properties:
+      RouteTableId: !Ref WSO2EnvPublicRouteTable1
+      SubnetId: !Ref WSO2EnvPublicSubnet1
+  subnetroute2:
+      Type: 'AWS::EC2::SubnetRouteTableAssociation'
+      Properties:
+        RouteTableId: !Ref WSO2EnvPublicRouteTable1
+        SubnetId: !Ref WSO2EnvPublicSubnet2
+  subnetroute3:
+    Type: 'AWS::EC2::SubnetRouteTableAssociation'
+    Properties:
+      RouteTableId: !Ref WSO2EnvRouteTable2
+      SubnetId: !Ref WSO2EnvPrivateSubnet1
+  route1:
+    Type: 'AWS::EC2::Route'
+    Properties:
+      DestinationCidrBlock: 0.0.0.0/0
+      RouteTableId: !Ref WSO2EnvPublicRouteTable1
+      GatewayId: !Ref WSO2EnvInternetGateway
+    DependsOn: gw1
+  route2:
+    Type: 'AWS::EC2::Route'
+    Properties:
+      DestinationCidrBlock: 0.0.0.0/0
+      RouteTableId: !Ref WSO2EnvRouteTable2
+      NatGatewayId: !Ref WSO2EnvNATGateway
+  dchpassoc1:
+    Type: 'AWS::EC2::VPCDHCPOptionsAssociation'
+    Properties:
+      VpcId: !Ref WSO2EnvVPC
+      DhcpOptionsId: !Ref WSO2EnvDHCPOptions
+
+  #########################
+  # Create security groups
+  #########################
+  WSO2EnvInstanceSecurityGroup:
+    Type: 'AWS::EC2::SecurityGroup'
+    Properties:
+      GroupDescription: Security group for IS EC2 compute instances
+      VpcId: !Ref WSO2EnvVPC
+      SecurityGroupIngress:
+      - IpProtocol: tcp
+        FromPort: 0
+        ToPort: 65535
+        CidrIp: 0.0.0.0/0
+      SecurityGroupEgress:
+      - IpProtocol: -1
+        CidrIp: 0.0.0.0/0
+  WSO2EnvLBSecurityGroup:
+    Type: 'AWS::EC2::SecurityGroup'
+    Properties:
+      GroupDescription: Security group for IS ALB
+      VpcId: !Ref WSO2EnvVPC
+      SecurityGroupIngress:
+      - IpProtocol: tcp
+        FromPort: 9763
+        ToPort: 9763
+        CidrIp: 0.0.0.0/0
+      - IpProtocol: tcp
+        FromPort: 8080
+        ToPort: 8080
+        CidrIp: 0.0.0.0/0
+      - IpProtocol: tcp
+        FromPort: 8081
+        ToPort: 8081
+        CidrIp: 0.0.0.0/0
+      - IpProtocol: tcp
+        FromPort: 80
+        ToPort: 80
+        CidrIp: 0.0.0.0/0
+      - IpProtocol: tcp
+        FromPort: 443
+        ToPort: 443
+        CidrIp: 0.0.0.0/0
+      SecurityGroupEgress:
+      - IpProtocol: -1
+        CidrIp: 0.0.0.0/0
+  WSO2EnvDBSecurityGroup:
+      Type: 'AWS::EC2::SecurityGroup'
+      Properties:
+        GroupDescription: Security group for IS RDS
+        VpcId: !Ref WSO2EnvVPC
+        SecurityGroupIngress:
+        - IpProtocol: tcp
+          FromPort: 3306
+          ToPort: 3306
+          CidrIp: 0.0.0.0/0
```

Comment:
Isn't it better to restrict this to connections from the instance security group?